rs/nns/governance/src/proposals/update_canister_settings.rs

```diff
@@ -31,6 +36,11 @@ impl UpdateCanisterSettings {
             .ok_or(invalid_proposal_error("Canister ID is required"))?;
         let canister_id = CanisterId::try_from(canister_principal_id)
             .map_err(|_| invalid_proposal_error("Invalid canister ID"))?;
+        if canister_id == ROOT_CANISTER_ID {
+            return Err(invalid_proposal_error(
```

Comment:
No change.

What if root were its own controller? IIRC, a canister can be its own controller. Not sure what happens if a canister that controls itself tells the management canister to update its own settings, but my guess is that it would "just work". This would simplify things.

tangent: I'm less sure that upgrading would work if root controlled itself...

Reply:
Interesting question. Now I'm curious what would happen for both `update_settings` and `install_code`. I think for the 2 management canister calls, I can't think of a reason why it wouldn't work. However, for `install_code` we usually stop before upgrading, and that should be impossible for self-controlled canisters.

Anyway, I don't suppose this should be blocking since letting root be self-controlled will be a much more significant change.